src/gcodecommands.cpp

```diff
@@ -91,6 +91,13 @@ QString toString(GCommands gcode)
 QString toCommand(GCommands gcode, const QString &value1)
 {
     switch (gcode) {
+    case G0: {
+        if (value1.isEmpty()) {
+            return QStringLiteral("G0");
+        } else {
+            return QStringLiteral("G0 %1").arg(value1.toUpper());
```

Comment:
Check if the argument is valid, if not scream something about it in qdebug.
value must have Xnnn or Ynnn [Link](http://reprap.org/wiki/G-code#G0_.26_G1:_Move)